MLIntegration/android/app/src/main/res/xml/config.xml

```diff
@@ -6,5 +6,9 @@
     <param name="android-package" value="com.android.plugins.Permissions"/>
   </feature>
 
+  <feature name="Device">
+    <param name="android-package" value="org.apache.cordova.device.Device"/>
+  </feature>
+
   
 </widget>
```

Comment:
It would be convenient to add EOF line break, so we're POSIX compliant. You can set up your IDE to alway do that on every edited file :)

Reply:
@jesusdiez This is a self-generated file by ionic, do I still have to add it?

Reply:
It's strange that a framework like ionic doesn't do things well. My recommendation is to autoconfigure your IDE to do it automatically when you edit the file, so in this case, as you're editing it, your changes include the EOF line break fix. You shouldn't have to recheck every automatically generated file to make it POSIX compliant anyway.

Some literature, FYI :)
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline